### PR TITLE
WDIO-Mocha Assignment

### PR DESCRIPTION
**Pull request by madhumitha-bs**
Latest results from Browserstack Automate: https://automate.browserstack.com/builds/63f304fa870dfbc9fc3b502ce167a3e4c5a42af6/sessions/3aa84dff03f9354728d2bfd02e03bd0bf809535c?auth_token=4346f0dbe044962f1c63ba874b56957d8793e38f312fb7e68def8c780320a73d

### CODE REVIEW

File: wdio-mocha-practice-assignment/test/pageobjects/home.page.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+     * overwrite specific options to adapt it to page object
+     */
+    public open () {
+        return super.open('');
+    }
+
+    public async searchProduct(productName:string){
+        const searchBtn= await this.searchButton;
+        await searchBtn.waitForDisplayed();
+        browser.pause
```
Comment (Sameer-Dhole) [inline wdio-mocha-practice-assignment/test/pageobjects/home.page.ts]: avoid explicit pauses , add awaits if using pauses

File: wdio-mocha-practice-assignment/wdio.conf.ts [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+            'sessionName': 'Session 5.11',
+            'local': false,
+            'networkLogs': true,
+            'video': true, 
+            'consoleLogs': 'verbose',
+            'seleniumVersion': '3.141.59',
+            'resolution': '1920x1080' 
+        }
+      }],
+    user:  'madhumithavenkat_ZxvnWs',
```
Comment (Sameer-Dhole) [inline wdio-mocha-practice-assignment/wdio.conf.ts]: fetch this from env variables and don't commit values

File: wdio-mocha-practice-assignment/wdio.conf.ts [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+    //
+    
+    capabilities: [{
+        'bstack:options': {
+            'os': 'Windows',
+            'osVersion': '10',
+            'browserName': 'Chrome',
+            'browserVersion': 'latest',
+            'projectName': 'My WebdriverIO Mocha Test Project',
+            'buildName': 'Build 5 for WebdriverIO Mocha ',
```
Comment (Sameer-Dhole) [inline wdio-mocha-practice-assignment/wdio.conf.ts]: make this dynamic

File: wdio-mocha-practice-assignment/test/specs/test.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import HomePage from '../pageobjects/home.page.js'
+import ResultsPage from '../pageobjects/results.page.js'
+
+describe('Verify the user is able to search a product', () => {
+    let productToSearch:string
+    
+    before(async () => {
+        // Runs before all tests in the suite
+        await HomePage.open()
+        productToSearch = "Laptop"
```
Comment (Sameer-Dhole) [inline wdio-mocha-practice-assignment/test/specs/test.e2e.ts]: use for each loop to run multiple examples instead od just laptop

File: wdio-mocha-practice-assignment/test/specs/test.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    });
+
+    it('the user should see the search product on the results page', async() =>{
+        await HomePage.searchProduct(productToSearch)
+        const spanText = await ResultsPage.getSearchResultSpan()
+        expect(spanText).toContain("Showing")
+        expect(spanText).toContain(" results for \"Laptop\"")
+    })
+
+    after(async () => {
```
Comment (Sameer-Dhole) [inline wdio-mocha-practice-assignment/test/specs/test.e2e.ts]: no need for this for only console.log

Comment (Sameer-Dhole) [reply wdio-mocha-practice-assignment/test/specs/test.e2e.ts]: move this to after block in conf

**Comment by madhumitha-bs** [comment]
@Sameer-Dhole I have addressed the comments for the PR. Latest build results: https://automate.browserstack.com/dashboard/v2/public-build/NVF4VU5tN21PVkFVSzF5TTB0d2luTzEzMkNQSUNud1dmSnViZWRJYkFJNTZrL3hGN1RyV0JjVUc5c1RTQVBNOEhtRTdVc1k0YlNHZ2Z0ZS9ENjhBV0E9PS0tQzlpWEx2MThhYWFZUGpwcTk0RG1wdz09--f24fdc466161293434aeefb68bea0178198a4c8a

File: wdio-mocha-practice-assignment/test/pageobjects/results.page.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  public get getSpanText(): ChainablePromiseElement<WebdriverIO.Element> {
+    return $(".BUOuZu");
+  }
+
+  public async getSearchResultSpan(): Promise<string> {
+    const spanText = await this.getSpanText;
+    await spanText.waitForDisplayed();
+    return spanText.getText();
```
Comment (Sameer-Dhole) [inline wdio-mocha-practice-assignment/test/pageobjects/results.page.ts]: use awaits on these methods

**Comment by Sameer-Dhole** [comment]
LGTM for wdio mocha and jenkins

**Pull request merged**